### ISSUE TITLE
A (potential bug) in data module

### ISSUE THREAD

**Issue by gpengzhi** [asyml/texar-pytorch#234]
Run `seq2seq_attn` example.

```python
python prepare_data.py --data toy_copy

python seq2seq_attn.py --config-model config_model --config-data config_toy_copy
```
Have the following error. I believe there was no such error before.

```
step=0, loss=36.4250
step=50, loss=22.1697
step=100, loss=13.6045
step=150, loss=6.7436
step=200, loss=4.6603
step=250, loss=2.5102
step=300, loss=0.2983
Traceback (most recent call last):
  File "seq2seq_attn.py", line 180, in <module>
    main()
  File "seq2seq_attn.py", line 166, in main
    _train_epoch()
  File "seq2seq_attn.py", line 131, in _train_epoch
    for batch in data_iterator:
  File "/Users/pengzhi.gao/Desktop/my_git/texar-pytorch/texar/torch/data/data/data_iterators.py", line 426, in __next__
    not self.dataset.hparams.allow_smaller_final_batch):
AttributeError: '_SPDataLoaderIter' object has no attribute 'dataset'
```

**Comment by huzecong** [comment]
Could you confirm which PyTorch version you're using? `_SPDataLoaderIter` is something we defined but only exists in PyTorch versions 1.2+, and it should be tested thoroughly in unit tests (`SP` stands for "single process" so it's indeed the most trivial case).

**Comment by huzecong** [comment]
Hmm, it's probably a compatibility issue with PyTorch 1.3. They've renamed variables in their data modules so everything is private and `dataset` became `_dataset`. I'm planning on rewriting the core code of data module so I might as well fix it with #225.

**Comment by gpengzhi** [comment]
Ah, I see. I think it is a compatibility issue with PyTorch 1.3.0.
